File: algorithms/[485]最大连续 1 的个数/solution.py
# topics = ["数组"]

from typing import List


class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        # 最大连续 1 的个数
        res = 0
        # 当前连续 1 的个数
        count = 0

        for num in nums:
            if num == 0:
                res = max(res, count)
                count = 0
            else:
                count += 1

        return max(res, count)

File: algorithms/[485]最大连续 1 的个数/solution.ts
// topics = ["数组"]

function findMaxConsecutiveOnes(nums: number[]): number {
  const n = nums.length;
  let i = 0;
  let j = 0;
  let res = 0;

  while (j < n) {
    if (nums[j] == 0) {
      res = Math.max(res, j - i);
      i = j + 1;
    }
    j += 1;
  }

  return Math.max(res, j - i);
}
